watch-library/hardware/watch/watch_buzzer.c

```diff
@@ -90,6 +90,15 @@ void watch_buzzer_play_sequence(int8_t *note_sequence, void (*callback_on_end)(v
     _tc3_start();
 }
 
+uint16_t sequence_length(int8_t *sequence) {
+    uint16_t result = 0;
+    int i = 0;
+    while (sequence[i++]) {
```

Comment:
Personally I'm not a fan of this iteration style (incrementing sequence both in the while condition and the body).

I would prefer:
```
uint16_t sequence_length(int8_t *sequence) {
    uint16_t result = 0;

    while (*sequence != 0){
        result += *(sequence + 1);
        sequence += 2;
    };

    return result;
}
```

Reply:
Yeah, I agree with that. Pushed a update, and also moved this function around so it's accessible to the simulator as well as hardware. That commit is untested, though, and it'll probably be a little while before I open up my watch again, if someone else wants to test it that'd be great.